Code/Mantid/Framework/Geometry/inc/MantidGeometry/Instrument/Parameter.h

```diff
@@ -91,7 +91,13 @@ class MANTID_GEOMETRY_DLL Parameter {
   /// Returns the parameter value of type T if the parameter has type
   /// ParameterType<T>
   template <class T> const T &value();
-
+  /// set description:
+  virtual void setDescription(const std::string &source)
+  {m_description.assign(source);}
+  /// get description
+  virtual const std::string & getDescription()const{return m_description;}
+  /// get short description (tooltip)
+  virtual std::string getTooltip()const;
```

Comment:
I think a better name for `getToolTip` would be `getShortDescription`. It doesn't have to be used for a tool tip, that's just a GUI thing.